packages/frameworks/vue/src/presence/use-presence.ts

```diff
@@ -18,20 +18,19 @@ export interface UsePresenceProps extends Optional<presence.Context, 'present'>
 
 export type UsePresenceReturn = ReturnType<typeof usePresence>
 
-export const usePresence = (props: UsePresenceProps, emit: CallableFunction) => {
-  const context = ref(props)
+export const usePresence = (props: MaybeRef<UsePresenceProps>, emit: CallableFunction) => {
+  const context = computed(() => ({
+    ...toValue(props),
+    // FIXME: we have to declare here, because zag did not set `onExitComplete` during initialization.
+    // FIX PR: https://github.com/chakra-ui/zag/pull/1124, maybe we can move `onExitComplete` to init later.
+    onExitComplete: () => {
+      emit('exit-complete')
+    },
+  }))
   const wasEverPresent = ref(false)
   const nodeRef = ref<VNodeRef | null>(null)
 
-  const [state, send] = useMachine(
-    presence.machine({
-      ...context.value,
-      onExitComplete: () => {
```

Comment:
If zag/presence set initial context during the initialization phase, we can back this.